src/imap/mod.rs

```diff
@@ -1070,11 +1070,18 @@ impl Imap {
         })
     }
 
-    pub fn ensure_configured_folders(&self, context: &Context, create_mvbox: bool) -> Result<()> {
+    pub fn ensure_configured_folders(
+        &self,
+        context: &Context,
+        create_mvbox: bool,
+        force_reconfigure: bool,
+    ) -> Result<()> {
         let folders_configured = context
             .sql
             .get_raw_config_int(context, "folders_configured");
-        if folders_configured.unwrap_or_default() >= DC_FOLDERS_CONFIGURED_VERSION {
+        if !force_reconfigure
```

Comment:
Instead of `force_reconfigure` argument, maybe split this function into `configure_folders`, which configures unconditionally, and `ensure_configured_folders`, which checks the condition and calls `configure_folders` if needed?

Reply:
yip, this will be more readable. thanks for the tip.